src/realm/sync/client.hpp

```diff
@@ -333,6 +333,10 @@ class Session {
         /// changeset data in a single integration attempt.
         size_t flx_bootstrap_batch_size_bytes = 1024 * 1024;
 
+        /// Contains the original PBS partition value from before the migration -
+        /// empty if not migrated
+        std::optional<std::string_view> migrated_partition;
```

Comment:
Should this really just be a std::shared_ptr<MigrationStore> and then the sync client can just query this as needed?

Reply:
Only the original string value (as an optional) is necessary and I didn't want to extend the lifetime of the migration_store beyond `SyncSession`. I can change it if you think it would be better.